packages/context/src/context.ts

```diff
@@ -211,18 +363,16 @@ export class Context {
    * binding is found, an error will be thrown.
    *
    * @param key Binding key
+   * @param optional If set to true, the method will return `undefined` instead
+   * of throwing an error if the binding key is not found
    */
-  getBinding(key: string): Binding {
+  getBinding(key: string, optional: boolean = false): Binding | undefined {
```

Comment:
Boolean arguments are evil - they are difficult to understand at call site, they are difficult to extend in the future as the number of flags grows.

Please use an options object with named properties instead.

```ts
ctx.getBinding(key, {optional: true});
```